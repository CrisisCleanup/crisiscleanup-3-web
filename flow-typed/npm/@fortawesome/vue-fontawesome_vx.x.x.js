// flow-typed signature: 88a7e004dad0f95b883e0a7cec99db76
// flow-typed version: <<STUB>>/@fortawesome/vue-fontawesome_v^0.1.10/flow_v0.129.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@fortawesome/vue-fontawesome'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@fortawesome/vue-fontawesome' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@fortawesome/vue-fontawesome/examples/vue-cli-webpack/build/build' {
  declare module.exports: any;
}

declare module '@fortawesome/vue-fontawesome/examples/vue-cli-webpack/build/check-versions' {
  declare module.exports: any;
}

declare module '@fortawesome/vue-fontawesome/examples/vue-cli-webpack/build/utils' {
  declare module.exports: any;
}

declare module '@fortawesome/vue-fontawesome/examples/vue-cli-webpack/build/vue-loader.conf' {
  declare module.exports: any;
}

declare module '@fortawesome/vue-fontawesome/examples/vue-cli-webpack/build/webpack.base.conf' {
  declare module.exports: any;
}

declare module '@fortawesome/vue-fontawesome/examples/vue-cli-webpack/build/webpack.dev.conf' {
  declare module.exports: any;
}

declare module '@fortawesome/vue-fontawesome/examples/vue-cli-webpack/build/webpack.prod.conf' {
  declare module.exports: any;
}

declare module '@fortawesome/vue-fontawesome/examples/vue-cli-webpack/config/dev.env' {
  declare module.exports: any;
}

declare module '@fortawesome/vue-fontawesome/examples/vue-cli-webpack/config' {
  declare module.exports: any;
}

declare module '@fortawesome/vue-fontawesome/examples/vue-cli-webpack/config/prod.env' {
  declare module.exports: any;
}

declare module '@fortawesome/vue-fontawesome/examples/vue-cli-webpack/src/main' {
  declare module.exports: any;
}

declare module '@fortawesome/vue-fontawesome/index.es' {
  declare module.exports: any;
}

declare module '@fortawesome/vue-fontawesome/rollup.config' {
  declare module.exports: any;
}

declare module '@fortawesome/vue-fontawesome/src/components/__fixtures__/helpers' {
  declare module.exports: any;
}

declare module '@fortawesome/vue-fontawesome/src/components/__fixtures__/icons' {
  declare module.exports: any;
}

declare module '@fortawesome/vue-fontawesome/src/components/__tests__/FontAwesomeIcon.test' {
  declare module.exports: any;
}

declare module '@fortawesome/vue-fontawesome/src/components/__tests__/FontAwesomeLayers.test' {
  declare module.exports: any;
}

declare module '@fortawesome/vue-fontawesome/src/components/__tests__/FontAwesomeLayersText.test' {
  declare module.exports: any;
}

declare module '@fortawesome/vue-fontawesome/src/components/FontAwesomeIcon' {
  declare module.exports: any;
}

declare module '@fortawesome/vue-fontawesome/src/components/FontAwesomeLayers' {
  declare module.exports: any;
}

declare module '@fortawesome/vue-fontawesome/src/components/FontAwesomeLayersText' {
  declare module.exports: any;
}

declare module '@fortawesome/vue-fontawesome/src/converter' {
  declare module.exports: any;
}

declare module '@fortawesome/vue-fontawesome/src' {
  declare module.exports: any;
}

declare module '@fortawesome/vue-fontawesome/src/logger' {
  declare module.exports: any;
}

declare module '@fortawesome/vue-fontawesome/src/utils' {
  declare module.exports: any;
}

// Filename aliases
declare module '@fortawesome/vue-fontawesome/examples/vue-cli-webpack/build/build.js' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome/examples/vue-cli-webpack/build/build'>;
}
declare module '@fortawesome/vue-fontawesome/examples/vue-cli-webpack/build/check-versions.js' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome/examples/vue-cli-webpack/build/check-versions'>;
}
declare module '@fortawesome/vue-fontawesome/examples/vue-cli-webpack/build/utils.js' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome/examples/vue-cli-webpack/build/utils'>;
}
declare module '@fortawesome/vue-fontawesome/examples/vue-cli-webpack/build/vue-loader.conf.js' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome/examples/vue-cli-webpack/build/vue-loader.conf'>;
}
declare module '@fortawesome/vue-fontawesome/examples/vue-cli-webpack/build/webpack.base.conf.js' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome/examples/vue-cli-webpack/build/webpack.base.conf'>;
}
declare module '@fortawesome/vue-fontawesome/examples/vue-cli-webpack/build/webpack.dev.conf.js' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome/examples/vue-cli-webpack/build/webpack.dev.conf'>;
}
declare module '@fortawesome/vue-fontawesome/examples/vue-cli-webpack/build/webpack.prod.conf.js' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome/examples/vue-cli-webpack/build/webpack.prod.conf'>;
}
declare module '@fortawesome/vue-fontawesome/examples/vue-cli-webpack/config/dev.env.js' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome/examples/vue-cli-webpack/config/dev.env'>;
}
declare module '@fortawesome/vue-fontawesome/examples/vue-cli-webpack/config/index' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome/examples/vue-cli-webpack/config'>;
}
declare module '@fortawesome/vue-fontawesome/examples/vue-cli-webpack/config/index.js' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome/examples/vue-cli-webpack/config'>;
}
declare module '@fortawesome/vue-fontawesome/examples/vue-cli-webpack/config/prod.env.js' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome/examples/vue-cli-webpack/config/prod.env'>;
}
declare module '@fortawesome/vue-fontawesome/examples/vue-cli-webpack/src/main.js' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome/examples/vue-cli-webpack/src/main'>;
}
declare module '@fortawesome/vue-fontawesome/index.es.js' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome/index.es'>;
}
declare module '@fortawesome/vue-fontawesome/index' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome'>;
}
declare module '@fortawesome/vue-fontawesome/index.js' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome'>;
}
declare module '@fortawesome/vue-fontawesome/rollup.config.js' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome/rollup.config'>;
}
declare module '@fortawesome/vue-fontawesome/src/components/__fixtures__/helpers.js' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome/src/components/__fixtures__/helpers'>;
}
declare module '@fortawesome/vue-fontawesome/src/components/__fixtures__/icons.js' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome/src/components/__fixtures__/icons'>;
}
declare module '@fortawesome/vue-fontawesome/src/components/__tests__/FontAwesomeIcon.test.js' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome/src/components/__tests__/FontAwesomeIcon.test'>;
}
declare module '@fortawesome/vue-fontawesome/src/components/__tests__/FontAwesomeLayers.test.js' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome/src/components/__tests__/FontAwesomeLayers.test'>;
}
declare module '@fortawesome/vue-fontawesome/src/components/__tests__/FontAwesomeLayersText.test.js' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome/src/components/__tests__/FontAwesomeLayersText.test'>;
}
declare module '@fortawesome/vue-fontawesome/src/components/FontAwesomeIcon.js' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome/src/components/FontAwesomeIcon'>;
}
declare module '@fortawesome/vue-fontawesome/src/components/FontAwesomeLayers.js' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome/src/components/FontAwesomeLayers'>;
}
declare module '@fortawesome/vue-fontawesome/src/components/FontAwesomeLayersText.js' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome/src/components/FontAwesomeLayersText'>;
}
declare module '@fortawesome/vue-fontawesome/src/converter.js' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome/src/converter'>;
}
declare module '@fortawesome/vue-fontawesome/src/index' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome/src'>;
}
declare module '@fortawesome/vue-fontawesome/src/index.js' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome/src'>;
}
declare module '@fortawesome/vue-fontawesome/src/logger.js' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome/src/logger'>;
}
declare module '@fortawesome/vue-fontawesome/src/utils.js' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome/src/utils'>;
}
