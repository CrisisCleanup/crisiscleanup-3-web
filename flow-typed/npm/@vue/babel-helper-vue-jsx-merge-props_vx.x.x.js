// flow-typed signature: d7a23c53f4091ef2bf1df8afa5eb43c4
// flow-typed version: <<STUB>>/@vue/babel-helper-vue-jsx-merge-props_v^1.0.0/flow_v0.131.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@vue/babel-helper-vue-jsx-merge-props'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@vue/babel-helper-vue-jsx-merge-props' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@vue/babel-helper-vue-jsx-merge-props/dist/helper' {
  declare module.exports: any;
}

// Filename aliases
declare module '@vue/babel-helper-vue-jsx-merge-props/dist/helper.js' {
  declare module.exports: $Exports<'@vue/babel-helper-vue-jsx-merge-props/dist/helper'>;
}
