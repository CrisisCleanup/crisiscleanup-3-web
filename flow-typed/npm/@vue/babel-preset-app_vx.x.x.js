// flow-typed signature: 6ba91a92845d9a556a5bcdcc4f033fed
// flow-typed version: <<STUB>>/@vue/babel-preset-app_v^4.3.1/flow_v0.129.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@vue/babel-preset-app'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@vue/babel-preset-app' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@vue/babel-preset-app/polyfillsPlugin' {
  declare module.exports: any;
}

// Filename aliases
declare module '@vue/babel-preset-app/index' {
  declare module.exports: $Exports<'@vue/babel-preset-app'>;
}
declare module '@vue/babel-preset-app/index.js' {
  declare module.exports: $Exports<'@vue/babel-preset-app'>;
}
declare module '@vue/babel-preset-app/polyfillsPlugin.js' {
  declare module.exports: $Exports<'@vue/babel-preset-app/polyfillsPlugin'>;
}
