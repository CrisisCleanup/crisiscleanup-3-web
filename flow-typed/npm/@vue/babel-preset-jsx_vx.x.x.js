// flow-typed signature: eeecaeff29e4a3a6c63c4fe7e16a0eb3
// flow-typed version: <<STUB>>/@vue/babel-preset-jsx_v^1.1.2/flow_v0.131.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@vue/babel-preset-jsx'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@vue/babel-preset-jsx' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@vue/babel-preset-jsx/dist/plugin.cjs' {
  declare module.exports: any;
}

// Filename aliases
declare module '@vue/babel-preset-jsx/dist/plugin.cjs.js' {
  declare module.exports: $Exports<'@vue/babel-preset-jsx/dist/plugin.cjs'>;
}
