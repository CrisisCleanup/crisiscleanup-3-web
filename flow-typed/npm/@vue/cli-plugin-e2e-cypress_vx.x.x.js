// flow-typed signature: 122252e4d027555936ded476aff636b4
// flow-typed version: <<STUB>>/@vue/cli-plugin-e2e-cypress_v^4.4.6/flow_v0.129.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@vue/cli-plugin-e2e-cypress'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@vue/cli-plugin-e2e-cypress' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@vue/cli-plugin-e2e-cypress/generator' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/_eslintrc' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/plugins' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/specs/test' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/support/commands' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/support' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-e2e-cypress/ui' {
  declare module.exports: any;
}

// Filename aliases
declare module '@vue/cli-plugin-e2e-cypress/generator/index' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress/generator'>;
}
declare module '@vue/cli-plugin-e2e-cypress/generator/index.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress/generator'>;
}
declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/_eslintrc.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/_eslintrc'>;
}
declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/plugins/index' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/plugins'>;
}
declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/plugins/index.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/plugins'>;
}
declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/specs/test.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/specs/test'>;
}
declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/support/commands.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/support/commands'>;
}
declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/support/index' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/support'>;
}
declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/support/index.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/support'>;
}
declare module '@vue/cli-plugin-e2e-cypress/index' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress'>;
}
declare module '@vue/cli-plugin-e2e-cypress/index.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress'>;
}
declare module '@vue/cli-plugin-e2e-cypress/ui.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress/ui'>;
}
