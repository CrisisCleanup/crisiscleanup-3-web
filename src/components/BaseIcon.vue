<template>
  <div class="flex items-center base-icon">
    <img :class="styles" :src="iconMap[type]" :alt="alt" :title="alt" />
    <slot></slot>
  </div>
</template>

<script>
/* eslint-disable global-require */
const iconMap = {
  dashboard: require('@/assets/icons/dashboard.svg'),
  cases: require('@/assets/icons/cases.svg'),
  print: require('@/assets/icons/print-big.svg'),
  phone: require('@/assets/icons/phone.svg'),
  share: require('@/assets/icons/share-big.svg'),
  download: require('@/assets/icons/download.svg'),
  active: require('@/assets/icons/active.svg'),
  edit: require('@/assets/icons/edit.svg'),
  flag: require('@/assets/icons/flag.svg'),
  'flag-filled': require('@/assets/icons/flag-filled.svg'),
  search: require('@/assets/icons/search.svg'),
  table: require('@/assets/icons/table.svg'),
  map: require('@/assets/icons/notactive.svg'),
  info: require('@/assets/icons/inform.svg'),
  trash: require('@/assets/icons/delete.svg'),
  organization: require('@/assets/icons/my-organization.svg'),
  history: require('@/assets/icons/history.svg'),
  cancel: require('@/assets/icons/big.svg'),
  help: require('@/assets/icons/help.svg'),
  up: require('@/assets/icons/up.svg'),
  down: require('@/assets/icons/down.svg'),
  updown: require('@/assets/icons/updown.svg'),
  'go-case': require('@/assets/icons/replace-case.svg'),
  'map-buffer': require('@/assets/icons/map-buffer.svg'),
  'map-circle': require('@/assets/icons/map-circle.svg'),
  'map-poly': require('@/assets/icons/map-poly.svg'),
  'map-rect': require('@/assets/icons/map-rect.svg'),
  'map-sweep': require('@/assets/icons/map-sweep.svg'),
  'map-undo': require('@/assets/icons/map-undo.svg'),
  'map-redo': require('@/assets/icons/map-redo.svg'),
};
/* eslint-enable global-require */

export default {
  name: 'BaseIcon',
  props: {
    type: {
      type: String,
      default: '',
    },
    alt: {
      type: String,
      default: '',
    },
    size: {
      type: String,
      default: '',
    },
  },
  data() {
    return {
      iconMap,
      styles: {
        'ccu-icon': true,
        'cursor-pointer': true,
        large: this.size === 'large',
        medium: this.size === 'medium',
        small: this.size === 'small',
        xs: this.size === 'xs',
        xl: this.size === 'xl',
      },
    };
  },
};
</script>

<style scoped>
.ccu-icon {
  height: 30px;
  width: 30px;
}
.ccu-icon.medium {
  height: 20px;
  width: 20px;
}
.ccu-icon.xl {
  height: 35px;
  width: 35px;
}
.ccu-icon.small {
  height: 16px;
  width: 16px;
}
.ccu-icon.xs {
  height: 10px;
  width: 10px;
}
.filter-gray {
  filter: invert(84%) sepia(0%) saturate(30%) hue-rotate(209deg)
    brightness(107%) contrast(90%);
}
.filter-yellow {
  filter: invert(92%) sepia(21%) saturate(3995%) hue-rotate(346deg)
    brightness(98%) contrast(106%);
}
</style>
