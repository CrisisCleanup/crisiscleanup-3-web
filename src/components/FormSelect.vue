<template>
  <div class="select__container">
    <resize-observer @notify="handleResize" />
    <v-select
      :input-id="inputIdSelector"
      :value="value"
      :options="options"
      :label="label"
      :components="{ OpenIndicator, Deselect }"
      :searchable="searchable"
      :multiple="multiple"
      :clearable="clearable"
      :disabled="disabled"
      class="form-select text-base"
      :class="[
        selectClasses,
        isInvalid && !value ? 'invalid' : '',
        floatLabel ? 'py-2 pt-3' : '',
      ]"
      :placeholder="placeholder"
      :reduce="(item) => (itemKey ? item[itemKey] : item)"
      :selectable="
        () =>
          !multiple || !limit || (multiple && limit > 0 && value.length < limit)
      "
      @input="onInput"
      @search:focus="open"
      @search:blur="onBlur"
      @search="(payload) => $emit('search', payload)"
      v-bind="$attrs"
    >
      <template #selected-option="option">
        <slot name="selected-option" :option="option" />
      </template>
      <template #option="option">
        <slot name="option" :option="option" />
      </template>
      <template v-if="required" #search="{ attributes, events }">
        <input
          ref="input"
          class="vs__search"
          :required="!value"
          :readonly="false"
          v-bind="attributes"
          v-on="events"
          :placeholder="placeholder"
        />
      </template>
    </v-select>
    <label
      :style="isFloated && floatStyle"
      v-if="floatLabel"
      ref="inputLabel"
      for="select-id"
    >
      <slot name="float-label" v-bind="{ isFloated }">
        {{ floatLabel }}
      </slot>
    </label>
  </div>
</template>

<script>
import { xor, kebabCase, isEmpty } from 'lodash';
export default {
  name: 'FormSelect',
  props: {
    searchable: {
      type: Boolean,
      default: false,
    },
    multiple: {
      type: Boolean,
      default: false,
    },
    disabled: {
      type: Boolean,
      default: false,
    },
    required: {
      type: Boolean,
      default: false,
    },
    clearable: {
      type: Boolean,
      default: true,
    },
    selectClasses: {
      type: String,
      default: '',
    },
    options: {
      type: Array,
      default: () => {
        return [];
      },
    },
    limit: {
      type: Number,
      default: 0,
    },
    placeholder: {
      type: String,
      default: '',
    },
    label: {
      type: String,
      default: '',
    },
    size: {
      type: String,
      default: 'large',
    },
    value: {
      type: null,
      default: null,
    },
    itemKey: {
      type: null,
      default: null,
    },
    indicatorIcon: {
      type: String,
      default: 'sort',
    },
    floatLabel: {
      type: String,
      default: '',
    },
  },
  data() {
    const iconSize = this.multiple ? 'xxs' : 'medium';
    const { indicatorIcon } = this;
    const cancelText = this.$t('actions.cancel');
    return {
      isFloated_: false,
      isEmpty_: true,
      isInvalid: false,
      baseHeight_: 0.0,
      currentHeight_: 0.0,
      floatDisplacement: 0,
      Deselect: {
        render() {
          return (
            <ccu-icon
              alt={cancelText}
              size={iconSize}
              class="mx-1 opacity-50"
              type="cancel"
            />
          );
        },
      },
      OpenIndicator: {
        render() {
          return (
            <font-awesome-icon
              size="sm"
              icon={indicatorIcon}
              class="mx-1 text-crisiscleanup-dark-400"
            />
          );
        },
      },
    };
  },
  computed: {
    inputIdSelector() {
      const idSpec = this.floatLabel ? this.floatLabel : '';
      return `select-id-${kebabCase(idSpec)}`;
    },
    inputRef() {
      if (this.$refs.input) return this.$refs.input;
      return document.getElementById(this.inputIdSelector);
    },
    heightMultiplier() {
      if (this.currentHeight_ === 0.0) return 1;
      this.$log.debug('current container height:', this.currentHeight_);
      const multi = this.currentHeight_ / this.baseHeight_;
      this.$log.debug('height multiplier change: ', multi);
      return multi;
    },
    floatStyle() {
      const displace = this.heightMultiplier * -12;
      return { transform: `translateY(${displace}px)` };
    },
    isEmpty() {
      if (this.value !== null) {
        return isEmpty(this.value);
      }
      return this.isEmpty_;
    },
    isFloated() {
      if (this.isFloated_ === true) {
        return true;
      }
      if (this.isEmpty === false) {
        return true;
      }
      return this.isFloated_;
    },
  },
  mounted() {
    this.id = this._uid;
    if (this.required) {
      this.$refs.input.addEventListener(
        'invalid',
        () => {
          this.isInvalid = true;
        },
        true,
      );
    }
    if (this.floatLabel) {
      this.inputRef.parentElement.classList.add('has-float');
      this.baseHeight_ = this.inputRef.parentElement.clientHeight;
    }
  },
  methods: {
    onInput(value) {
      this.$emit('input', value);
      this.isEmpty_ = value === null;
      if (value === null && this.floatLabel) {
        this.isFloated_ = false;
        this.$refs.inputLabel.classList.remove('focused');
      }
      if (this.multiple) {
        const current = this.value || [];
        this.$emit(
          'changed',
          xor(
            value,
            current.map((item) => (this.itemKey ? item[this.itemKey] : item)),
          ),
        );
      }
      if (this.required) {
        if (this.$refs.input.checkValidity()) {
          this.isInvalid = false;
        }
      }
    },
    onBlur() {
      if (this.floatLabel && this.isEmpty) {
        this.isFloated_ = false;
        this.$refs.inputLabel.classList.remove('focused');
      }
    },
    open() {
      if (this.floatLabel) {
        this.isFloated_ = true;
        this.$refs.inputLabel.classList.add('focused');
      }
      this.$nextTick(() => {
        const items = [].slice.call(
          document.querySelectorAll('.vs__dropdown-option'),
        );

        items.forEach((item) => {
          item.classList.remove('vs__dropdown-option--highlight');
        });

        const selected = items.find((item) => {
          if (this.value) {
            const currentLabel = this.label
              ? this.value[this.label]
              : this.value;
            return item.textContent.trim() === currentLabel;
          }
          return false;
        });

        if (selected) {
          selected.classList.add('vs__dropdown-option--highlight');
          document.querySelector('.vs__dropdown-menu').scrollTop =
            selected.offsetTop;
        }
      });
    },
    handleResize() {
      this.currentHeight_ = this.inputRef.parentElement.clientHeight;
      this.$emit('resize', this.currentHeight_);
    },
  },
};
</script>

<style>
@import '~vue-select/dist/vue-select.css';
.form-select .vs__dropdown-menu {
  border-radius: 0;
}
.form-select .vs__dropdown-toggle {
  height: 100%;
  border-radius: 0;
  border: none;
  .vs__selected-options.has-float {
    margin-top: 5px;
    margin-bottom: -5px;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
  }
}
.form-select .vs__search::placeholder {
  @apply text-crisiscleanup-dark-200;
}

.form-select {
  &.vs--single .has-float {
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
    span {
      @apply font-bold;
    }
  }
}

.form-select.invalid {
  @apply border border-crisiscleanup-red-100;
}

.vue-select-up.form-select .vs__dropdown-menu {
  top: auto !important;
  bottom: calc(100% - 3px);
}
</style>

<style lang="postcss">
.select {
  &__container {
    position: relative;
    > label {
      @apply text-h2 font-normal text-crisiscleanup-dark-400;
      position: absolute;
      top: 25%;
      bottom: 0;
      left: 1%;
      width: 100%;
      height: 100%;
      pointer-events: none;
      transition: transform 200ms easeInOutQuint, color 150ms easeInOutQuint,
        font-size 150ms easeInOutQuint;
      &.focused {
        @apply text-h4 font-normal text-crisiscleanup-dark-300;
      }
    }
  }
}
</style>
