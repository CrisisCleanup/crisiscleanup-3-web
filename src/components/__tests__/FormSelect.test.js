/**
 *
 * Tests for FormSelect
 *
 * Components
 */

import { mount } from '@vue/test-utils';
import vSelect from 'vue-select';
import { library } from '@fortawesome/fontawesome-svg-core';
import { fas } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome';
import BaseIcon from '@/components/BaseIcon.vue';
import FormSelect from '../FormSelect.vue';
library.add(fas);

const mountWithOptions = (options) =>
  mount(FormSelect, {
    stubs: {
      'v-select': vSelect,
      'font-awesome-icon': FontAwesomeIcon,
      'ccu-icon': BaseIcon,
      'resize-observer': true,
    },
    propsData: {
      searchable: true,
      placeholder: 'Some Value',
    },
    mocks: {
      $t: (key) => key,
    },
    ...options,
  });

describe('FormSelect', () => {
  it('should render correctly and match snapshot', () => {
    const wrapper = mountWithOptions();
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should match snapshot with props', () => {
    const wrapper = mountWithOptions();
    expect(wrapper.element).toMatchSnapshot();
  });
});
