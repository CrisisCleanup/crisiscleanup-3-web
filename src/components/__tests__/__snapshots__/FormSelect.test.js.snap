// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormSelect should match snapshot with props 1`] = `
<div
  class="v-select form-select text-base vs--single vs--searchable"
  dir="auto"
>
  <div
    class="vs__dropdown-toggle"
  >
    <div
      class="vs__selected-options"
    >
       
      <input
        aria-label="Search for option"
        autocomplete="off"
        class="vs__search"
        placeholder="Some Value"
        role="combobox"
        type="search"
      />
    </div>
     
    <div
      class="vs__actions"
    >
      <button
        class="vs__clear"
        style="display: none;"
        title="Clear selection"
        type="button"
      >
        <div
          class="flex items-center base-icon js-actions-cancel   mx-1 opacity-50"
        >
          <img
            alt="actions.cancel"
            class="ccu-icon cursor-pointer medium"
            src="IMAGE_MOCK"
            title="actions.cancel"
          />
           
        </div>
      </button>
       
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-sort fa-w-10 fa-sm mx-1 text-crisiscleanup-dark-400 vs__open-indicator"
        data-icon="sort"
        data-prefix="fas"
        focusable="false"
        role="presentation"
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          class=""
          d="M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41zm255-105L177 64c-9.4-9.4-24.6-9.4-33.9 0L24 183c-15.1 15.1-4.4 41 17 41h238c21.4 0 32.1-25.9 17-41z"
          fill="currentColor"
        />
      </svg>
       
      <div
        class="vs__spinner"
        style="display: none;"
      >
        Loading...
      </div>
    </div>
  </div>
   
  <!---->
</div>
`;

exports[`FormSelect should render correctly and match snapshot 1`] = `
<div
  class="v-select form-select text-base vs--single vs--searchable"
  dir="auto"
>
  <div
    class="vs__dropdown-toggle"
  >
    <div
      class="vs__selected-options"
    >
       
      <input
        aria-label="Search for option"
        autocomplete="off"
        class="vs__search"
        placeholder="Some Value"
        role="combobox"
        type="search"
      />
    </div>
     
    <div
      class="vs__actions"
    >
      <button
        class="vs__clear"
        style="display: none;"
        title="Clear selection"
        type="button"
      >
        <div
          class="flex items-center base-icon js-actions-cancel   mx-1 opacity-50"
        >
          <img
            alt="actions.cancel"
            class="ccu-icon cursor-pointer medium"
            src="IMAGE_MOCK"
            title="actions.cancel"
          />
           
        </div>
      </button>
       
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-sort fa-w-10 fa-sm mx-1 text-crisiscleanup-dark-400 vs__open-indicator"
        data-icon="sort"
        data-prefix="fas"
        focusable="false"
        role="presentation"
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          class=""
          d="M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41zm255-105L177 64c-9.4-9.4-24.6-9.4-33.9 0L24 183c-15.1 15.1-4.4 41 17 41h238c21.4 0 32.1-25.9 17-41z"
          fill="currentColor"
        />
      </svg>
       
      <div
        class="vs__spinner"
        style="display: none;"
      >
        Loading...
      </div>
    </div>
  </div>
   
  <!---->
</div>
`;
